### PR TITLE
Holidays

### PR DESCRIPTION
**Pull request by frederic34**

### CODE REVIEW

File: htdocs/core/class/notify.class.php [hunk 1]
```diff
@@ -314,218 +317,346 @@ function send($notifcode, $object)
...
-			}
-		}
-		else
-		{
-	   		$error++;
-			$this->errors[]=$this->db->lasterror();
-			dol_syslog("Failed to get list of notification to send ".$this->db->lasterror(), LOG_ERR);
-	   		return -1;
-		}
+        if (! empty($object->socid)) {
```
Comment (eldy) [inline htdocs/core/class/notify.class.php]: If you add a test here, the second part that is a union to get target to users id will not be included.
If $object->socid is not defined, the test
$sql .= " AND s.rowid = ".$object->socid;
can be replaced with 
$sql .= " AND s.rowid = ".($object->socid>0?$object->socid:0);

File: htdocs/core/class/notify.class.php [hunk 2]
```diff
@@ -314,218 +317,346 @@ function send($notifcode, $object)
...
+		    else
+		    {
+	   		    $error++;
+			    $this->errors[]=$this->db->lasterror();
+			    dol_syslog("Failed to get list of notification to send ".$this->db->lasterror(), LOG_ERR);
+	   		    return -1;
+            }
+        }
+
+        // notify user and validator for holiday if subscribed to notification
```
Comment (eldy) [inline htdocs/core/class/notify.class.php]: Can you explain me what is effect of this code ?
When you go on page http://localhostgit/dolibarr_dev/htdocs/user/notify/card.php?id=12
a user can subscribe to event he want to register.
Does this means for holiday validation and approval, if he user register he will need to register to receive event ?

I don't like idea to have email for human resource module (holiday and expense report), that are modules where view depends on a supervisor link, that are on of off according to notification module that is a module dedicated to allow a user to register on events to get emails.

Comment (frederic34) [reply htdocs/core/class/notify.class.php]: notification - firefox developer edition

File: htdocs/core/class/notify.class.php (hunk 1 above)
Comment (frederic34) [reply htdocs/core/class/notify.class.php]: change done, thanks

**Comment by frederic34** [comment]
⚠️ wip I removed notifications, I need to improve holidays in agenda

**Pull request closed without merging**